apps/checkout-app/pages/customization.tsx

```diff
@@ -2,30 +2,32 @@ import { useRouter } from "next/router";
 import CustomizationDetails from "@/frontend/components/templates/CustomizationDetails";
 import { CustomizationSettingsValues } from "types/api";
 import {
-  usePublicMetadataQuery,
+  usePublicMetafieldsQuery,
   useUpdatePublicMetadataMutation,
 } from "@/graphql";
 import { getCommonErrors } from "@/frontend/utils";
 import { useCustomizationSettings } from "@/frontend/data";
 import { useAuthData } from "@/frontend/hooks/useAuthData";
 import { serverEnvVars } from "@/constants";
 import { mapPublicSettingsToMetadata } from "@/frontend/misc/mapPublicSettingsToMetadata";
-import { mapPublicMetadataToSettings } from "@/frontend/misc/mapPublicMetadataToSettings";
+import { mapPublicMetafieldsToSettings } from "@/frontend/misc/mapPublicMetafieldsToSettings";
+import { PublicSettingID } from "@/types/common";
 
 const Customization = () => {
   const router = useRouter();
   const { appId, isAuthorized } = useAuthData();
-  const [metadataQuery] = usePublicMetadataQuery({
+  const [metafieldsQuery] = usePublicMetafieldsQuery({
     variables: {
       id: appId || serverEnvVars.appId,
+      keys: ["customizations"] as PublicSettingID[number][],
```

Comment:
doesnt `PublicSettingID[number][]` === `PublicSettingID`?

Reply:
No, `PublicSettingID` requires all members of its array and they are readonly.